utils/build/build.sh

```diff
@@ -217,9 +217,27 @@ build() {
             else
 
                 # dd-trace-py compilation if required
-                if [[ $TEST_LIBRARY == python* ]] && [[ -d "binaries/dd-trace-py" ]]; then
+                if [[ $TEST_LIBRARY == python ]] && [[ -d "binaries/dd-trace-py" ]]; then
                     echo "Compiling dd-trace-py"
-                    docker run -v ./binaries/:/app -w /app ghcr.io/datadog/dd-trace-py/testrunner bash -c "pyenv global 3.11; pip wheel --no-deps -w . /app/dd-trace-py"
+                    
+                    # Choose Python version based on weblog variant
+                    case "$WEBLOG_VARIANT" in
+                        flask-poc|django-poc|fastapi|uds-flask|uwsgi-poc)
+                            PYTHON_VERSION="3.11"
+                            ;;
+                        django-py3.13)
+                            PYTHON_VERSION="3.13"
+                            ;;
+                        python3.12)
+                            PYTHON_VERSION="3.12"
+                            ;;
+                        *)
+                            PYTHON_VERSION="3.11"  # Default fallback
```

Comment:
Can you raise an error with a proper message? Otherwise, `install_ddtrace` will raise it, and it'll be harder to understand what's going on.